cmd/kube-proxy/app/server_others.go

```diff
@@ -183,29 +181,29 @@ func newProxyServer(
 	iptInterface = utiliptables.New(execer, primaryProtocol)
 
 	var ipt [2]utiliptables.Interface
-	dualStack := utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) && proxyMode != proxyModeUserspace
-	if dualStack {
+	dualStack := true // While we assume that node supports, we do further checks below
+
+	if proxyMode != proxyModeUserspace {
 		// Create iptables handlers for both families, one is already created
 		// Always ordered as IPv4, IPv6
 		if primaryProtocol == utiliptables.ProtocolIPv4 {
 			ipt[0] = iptInterface
 			ipt[1] = utiliptables.New(execer, utiliptables.ProtocolIPv6)
-
-			// Just because the feature gate is enabled doesn't mean the node
-			// actually supports dual-stack
-			if _, err := ipt[1].ChainExists(utiliptables.TableNAT, utiliptables.ChainPostrouting); err != nil {
-				klog.Warningf("No iptables support for IPv6: %v", err)
-				dualStack = false
-			}
 		} else {
 			ipt[0] = utiliptables.New(execer, utiliptables.ProtocolIPv4)
 			ipt[1] = iptInterface
 		}
-	}
-	if dualStack {
-		klog.V(0).InfoS("kube-proxy running in dual-stack mode", "ipFamily", iptInterface.Protocol())
-	} else {
-		klog.V(0).InfoS("kube-proxy running in single-stack mode", "ipFamily", iptInterface.Protocol())
+
+		// TODO: @khenidak - should we fail here?
+		// we can not tell if family failed == family used by cluster at this stage.
+		// MUST BE ADDRESSED PART OF KUBE-PROXY DUAL STACK STABLE
```

Comment:
I don't know what this comment means, but it is scary.  Are you asserting that kube-proxy is not dual-stack now?

Reply:
at this point we know that node does not support IPv6. But we can not tell if we need it or not. I will remove the comment. it is highly unlikely that this happens anyway.